pages/product/[code].js

```diff
@@ -102,69 +178,81 @@ export default function DetailPage() {
 
       {data && data.product ? (
         <>
-          <h2>{data.product.product_name} </h2>
-          {/* if user have not chosen additives show message */}
-          {additivesFromStorage.length > 0 ? (
-            <p>
-              Additive:
-              {filteredAdditives.length > 0 ? (
-                <span> ❌</span>
-              ) : (
-                <span> ✅</span>
-              )}
-            </p>
-          ) : (
-            <p>choose your additives</p>
-          )}
-          {/* if user have not chosen allergens show message */}
-          {allergensFromStorage.length > 0 ? (
-            <p>
-              Allergene:
-              {filteredAllergens.length > 0 ? (
-                <span> ❌</span>
+          <StyledAllCards>
+            <StyledProductCard>
+              <StyledProductName>
+                {data.product.product_name}{" "}
+              </StyledProductName>
+
+              {/* if user have not chosen additives show message */}
+              <StyledInsideCard>
+                <StyledImage
+                  width={1000}
+                  height={1000}
+                  src={data.product.image_front_url}
+                  alt={data.product.product_name}
+                />
+                <StyledCheck>
+                  {additivesFromStorage.length > 0 ? (
+                    <StyledPAdditives>
+                      Additive:
+                      {filteredAdditives.length > 0 ? (
+                        <span> ❌</span>
+                      ) : (
+                        <span> ✅</span>
+                      )}
+                    </StyledPAdditives>
+                  ) : (
+                    <p>keine Zusatzstoffe in den Einstellungen gewählt</p>
+                  )}
+                  {/* if user have not chosen allergens show message */}
+                  {allergensFromStorage.length > 0 ? (
+                    <StyledPAllergens>
+                      Allergene:
+                      {filteredAllergens.length > 0 ? (
+                        <span> ❌</span>
+                      ) : (
+                        <span> ✅</span>
+                      )}
+                    </StyledPAllergens>
+                  ) : (
+                    <p>keine Allergene in den Einstellunegn gewählt</p>
+                  )}
+                </StyledCheck>
+              </StyledInsideCard>
+
+              <p>Hersteller:{data.product.brands}</p>
+            </StyledProductCard>
+            {/* show allergens and additives */}
+            <StyledIngredientsCard>
+              <StyledAdditivesH3>Additive</StyledAdditivesH3>
+
+              {data.product && data.product.additives_original_tags ? (
+                data.product.additives_original_tags.map((additive) => (
+                  <Additives
+                    key={additive}
+                    additive={
+                      additives.tags.find((addi) => {
```

Comment:
I would rename `addi` to something more descriptive

Reply:
i will think about.but in this case it seems for me the logical way to not confuse myself with overusing the word additives in singular an plural.

Reply:
that is understandable. Using the word `tag` in place of `addi` could be a good solution, since you are going over an array of tags